qui/tray/devices.py

```diff
@@ -158,92 +128,98 @@ def toggle(self, menu_item):
             self.attach(menu_item)
 
     def attach(self, menu_item):
-        vm_name = menu_item.vm.name
-
-        if self.attached_item is not None:
+        if self.attached_items is not None:
```

Comment:
`self.attached_items` is never `None` (you initialize it with `[]`). But `self.detach` do handle not attached cases, so this condition could be dropped.